File: mvvp-arch/app/src/main/java/thaihn/android/demomvvm/screen/BaseActivity.java
package thaihn.android.demomvvm.screen;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    protected abstract int getLayoutResource();

    protected abstract void initVariables(Bundle saveInstanceState);

    protected abstract void initData(Bundle saveInstanceState);

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResource());
        initVariables(savedInstanceState);
        initData(savedInstanceState);
    }
}

File: clear-architect/presentation/src/main/java/thaihn/com/androidcleanarchitecture/base/BaseActivity.kt
package thaihn.com.androidcleanarchitecture.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import thaihn.com.androidcleanarchitecture.databinding.ActivityMainBinding

abstract class BaseActivity : AppCompatActivity() {

  abstract fun getLayoutResource(): Int

  abstract fun initVariable(savedInstanceState: Bundle?)

  abstract fun initData(savedInstanceState: Bundle?)

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(getLayoutResource())
    initVariable(savedInstanceState)
    initData(savedInstanceState)
  }
}